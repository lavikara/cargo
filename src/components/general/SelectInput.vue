<template>
  <div id="select-input">
    <div class="tw-relative tw-flex tw-flex-col">
      <label :for="label" class="tw-font-medium tw-whitespace-nowrap">
        {{ showLabel ? label : "" }}
      </label>
      <select
        ref="select"
        class="tw-w-full tw-bg-white tw-border tw-border-gray-border tw-rounded-2xl tw-cursor-pointer focus:tw-border-blue hover:tw-border-blue tw-px-4 tw-py-3 tw-mt-1"
        :class="{
          '!tw-bg-gray-border': bgColor === 'gray',
          '!tw-cursor-not-allowed': disabled,
          '!tw-rounded-tr-none !tw-rounded-br-none': leftSide,
          '!tw-rounded-tl-none !tw-rounded-bl-none': rightSide,
          'select-appearance': !appearance,
          '!tw-px-2 !tw-py-3': appearance,
        }"
        :name="name"
        :id="id"
        v-model="optionData"
        :required="required"
        :disabled="disabled"
        :readonly="readonly"
      >
        <option value="" disabled selected>
          {{ defaltOption }}
        </option>
        <option v-for="(item, index) in list" :value="item" :key="index">
          <p>
            {{ item }}
          </p>
        </option>
      </select>
      <svg
        v-if="!appearance"
        class="tw-absolute tw-right-2 tw-bottom-[1.3rem] tw-z-10"
        width="10"
        height="6"
        viewBox="0 0 10 6"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.99992 2.82901L7.82792 1.33514e-05L9.24292 1.41401L4.99992 5.65701L0.75692 1.41401L2.17192 1.33514e-05L4.99992 2.82901Z"
          fill="#4F5877"
        />
      </svg>
    </div>
  </div>
</template>

<script setup>
import { ref, watchEffect, watch, onMounted } from "vue";

const emit = defineEmits(["set"]);

const props = defineProps({
  label: { type: String, default: () => "", required: true },
  showLabel: { type: Boolean, default: () => false },
  required: { type: Boolean, default: () => true },
  readonly: { type: Boolean, default: () => false },
  disabled: { type: Boolean, default: () => false },
  clearOption: { type: Boolean, default: () => false },
  tableHeader: { type: Boolean, default: () => false },
  appearance: { type: Boolean, default: () => false },
  leftSide: { type: Boolean, default: () => false },
  rightSide: { type: Boolean, default: () => false },
  bgColor: { type: String, default: () => "" },
  name: { type: String, default: () => "" },
  id: { type: String, default: () => "" },
  defaltOption: { type: String, default: () => "" },
  list: { type: Array, default: () => [], required: true },
});

let select = ref();
let optionData = ref("");

onMounted(() => {
  optionData.value = "";
  emitOption(optionData.value);
});

const emitOption = (option) => {
  emit("set", option);
};

const setOptionData = (value) => {
  optionData.value = value;
  emitOption(value);
};

const clearOptionData = () => {
  optionData.value = "";
};

// watchEffect(() => setOptionData(optionData.value));
// watchEffect(() => setOptionData(props.option));
// watchEffect(() => clearOptionData(props.clearOption));
watch(
  () => props.clearOption,
  () => {
    clearOptionData(props.clearOption);
  }
);
watch(
  () => optionData.value,
  (newVal, oldVal) => {
    setOptionData(newVal);
  }
);
</script>

<style lang="scss" scoped></style>
