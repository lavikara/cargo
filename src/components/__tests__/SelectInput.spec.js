import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import SelectInput from "../general/SelectInput.vue";

describe("SelectInput", () => {
  it("renders properly", () => {
    const wrapper = mount(SelectInput);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
