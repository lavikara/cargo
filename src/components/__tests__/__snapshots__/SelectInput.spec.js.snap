// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectInput > renders properly 1`] = `
"<div id="select-input">
  <div class="tw-relative tw-flex tw-flex-col"><label for="" class="tw-font-medium tw-whitespace-nowrap"></label><select class="tw-w-full tw-bg-white tw-border tw-border-gray-border tw-rounded-2xl tw-cursor-pointer focus:tw-border-blue hover:tw-border-blue tw-px-4 tw-py-3 tw-mt-1 select-appearance" name="" id="" required="">
      <option disabled="" value=""></option>
    </select><svg class="tw-absolute tw-right-2 tw-bottom-[1.3rem] tw-z-10" width="10" height="6" viewBox="0 0 10 6" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path d="M4.99992 2.82901L7.82792 1.33514e-05L9.24292 1.41401L4.99992 5.65701L0.75692 1.41401L2.17192 1.33514e-05L4.99992 2.82901Z" fill="#4F5877"></path>
    </svg></div>
</div>"
`;

exports[`TextInput > renders properly 1`] = `
"<div data-v-9344f502="" id="text-input">
  <div data-v-9344f502="" class="tw-relative tw-flex tw-flex-col"><label data-v-9344f502="" for="" class="tw-font-medium tw-whitespace-nowrap"></label><input data-v-9344f502="" class="tw-w-full tw-bg-white tw-border tw-border-gray-border tw-rounded-2xl tw-transition-all tw-duration-300 focus:tw-border-blue hover:tw-border-blue tw-px-4 tw-py-3 tw-mt-1" type="text" placeholder="" required="" name="" id="">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <p data-v-9344f502="" class="tw-text-red tw-text-xs tw-absolute tw--bottom-4 tw-right-0"></p>
  </div>
</div>"
`;
